storage/replicate_queue.go

```diff
@@ -75,21 +75,35 @@ func (rq replicateQueue) shouldQueue(now proto.Timestamp, repl *Replica) (should
 		return
 	}
 
-	return rq.needsReplication(zone, repl, repl.Desc())
+	delta := rq.replicaDelta(zone, repl, repl.Desc())
+	if delta == 0 {
+		if log.V(1) {
+			log.Infof("%s has the correct number of nodes", repl)
+		}
+		return false, 0
+	}
+	if delta > 0 {
+		if log.V(1) {
+			log.Infof("%s needs to add %d nodes", repl, delta)
+		}
+		// For ranges which need additional replicas, increase the priority
+		return true, float64(delta + 10)
+	}
+	if log.V(1) {
+		log.Infof("%s needs to remove %d nodes", repl, 0-delta)
+	}
+	// For ranges which have too many replicas, priority is absolute value of
+	// the delta.
+	return true, float64(0 - delta)
```

Comment:
Can we skew towards ranges with less replicas from their quorum amount?
A 3 replica system missing one replica should have a higher priority than than a 5 replica system missing only one.  
Ben suggested using (quorum count)-(live nodes count) in the repair range pr #2336